insalan/tournament/views.py

```diff
@@ -309,8 +319,89 @@ class TeamDetails(generics.RetrieveUpdateDestroyAPIView):
 
     queryset = Team.objects.all().order_by("id")
     serializer_class = serializers.TeamSerializer
-    permission_classes = [permissions.IsAdminUser | ReadOnly]
+    permission_classes = [permissions.IsAdminUser | ReadOnlyPlusPatch]
+
+    def patch(self, request, *args, **kwargs):
+        """
+        Patch a team
+        """
+        user = request.user
+        data = request.data
+
+        if user is None or not user.is_authenticated:
+            raise PermissionDenied()
 
+        # get the team
+        team = Team.objects.get(id=kwargs["pk"])
+
+        # check if the user is registered in the team
+        player = Player.objects.filter(user=user, team=team)
+        manager = Manager.objects.filter(user=user, team=team)
+        if len(player) == 0 and len(manager) == 0:
+            return Response({
+                "team": _("Vous n'êtes pas inscrit dans cette équipe.")
+            }, status=status.HTTP_403_FORBIDDEN)
+
+        # check if the player is the team's captain or a manager
+        if len(manager) == 0 and team.get_players_id()[0] != player[0].id:
```

Comment:
Taking a look at the implementation, we have absolutely no guarantee that the order of the players is consistent!

Behind the scene, it resolves as an SQL request in an associative table PlayerId -> TeamId. Following PostgreSQL configuration *and* the database size, the returned order may be different from the insertion order. Even worse, it can change between two consecutive requests depending on a lot of external factors.

We should find a different way to define a Team's Captain

Reply:
agree, this was a quick'n dirty implementation, maybe, we should add a boolean field. This way, we could even implement a feature that allow changing Team's Captain